hudi-utilities/src/test/java/org/apache/hudi/utilities/TestHoodieSnapshotExporter.java

```diff
@@ -18,205 +18,144 @@
+    hdfsWriteClient.bulkInsert(recordsRDD, COMMIT_TIME);
+    hdfsWriteClient.close();
+
+    RemoteIterator<LocatedFileStatus> itr = dfs.listFiles(new Path(sourcePath), true);
+    while (itr.hasNext()) {
+      LOG.info(">>> Prepared test file: " + itr.next().getPath());
+    }
   }
 
   @After
-  public void cleanup() {
-    if (spark != null) {
-      spark.stop();
-    }
+  public void tearDown() throws Exception {
+    cleanupSparkContexts();
+    cleanupDFS();
+    cleanupTestDataGenerator();
   }
 
-  @Test
-  public void testSnapshotExporter() throws IOException {
-    // Insert Operation
-    List<String> records = DataSourceTestUtils.convertToStringList(dataGen.generateInserts("000", 100));
-    Dataset<Row> inputDF = spark.read().json(new JavaSparkContext(spark.sparkContext()).parallelize(records, 2));
-    inputDF.write().format("hudi")
-        .options(commonOpts)
-        .option(DataSourceWriteOptions.OPERATION_OPT_KEY(), DataSourceWriteOptions.INSERT_OPERATION_OPT_VAL())
-        .mode(SaveMode.Overwrite)
-        .save(basePath);
-    long sourceCount = inputDF.count();
-
-    HoodieSnapshotExporter hoodieSnapshotExporter = new HoodieSnapshotExporter();
-    hoodieSnapshotExporter.export(spark, cfg);
-
-    long targetCount = spark.read().json(outputPath).count();
-
-    assertTrue(sourceCount == targetCount);
-
-    // Test Invalid OutputFormat
-    cfg.outputFormat = "foo";
-    int isError = hoodieSnapshotExporter.export(spark, cfg);
-    assertTrue(isError == -1);
+  private HoodieWriteConfig getHoodieWriteConfig(String basePath) {
+    return HoodieWriteConfig.newBuilder()
+        .withPath(basePath)
+        .withEmbeddedTimelineServerEnabled(false)
+        .withSchema(HoodieTestDataGenerator.TRIP_EXAMPLE_SCHEMA)
+        .withParallelism(2, 2)
+        .withBulkInsertParallelism(2)
+        .forTable(TABLE_NAME)
+        .withIndexConfig(HoodieIndexConfig.newBuilder().withIndexType(IndexType.BLOOM).build())
+        .build();
   }
 
-  // for testEmptySnapshotCopy
-  public void init() throws IOException {
-    TemporaryFolder folder = new TemporaryFolder();
-    folder.create();
-    rootPath = "file://" + folder.getRoot().getAbsolutePath();
-    basePath = rootPath + "/" + HoodieTestUtils.RAW_TRIPS_TEST_NAME;
-    outputPath = rootPath + "/output";
-
-    final Configuration hadoopConf = HoodieTestUtils.getDefaultHadoopConf();
-    fs = FSUtils.getFs(basePath, hadoopConf);
-    HoodieTestUtils.init(hadoopConf, basePath);
+  @Test
+  public void testExportAsParquet() throws IOException {
+    HoodieSnapshotExporter.Config cfg = new Config();
+    cfg.sourceBasePath = sourcePath;
+    cfg.targetOutputPath = targetPath;
+    cfg.outputFormat = "parquet";
+    new HoodieSnapshotExporter().export(SparkSession.builder().config(jsc.getConf()).getOrCreate(), cfg);
+    assertEquals(NUM_RECORDS, sqlContext.read().parquet(targetPath).count());
+    assertTrue(dfs.exists(new Path(targetPath + "/_SUCCESS")));
```

Comment:
Does '/_SUCCESS' auto generated by hudi HoodieSnapshotExporte? I would only see '/_SUCCESS' in HoodieSnapshotCopier.

Reply:
@leesf Exporter seems has removed that logic for hudi case, though non-hudi case will do it automatically via spark. I'll ensure both create the success tag.